### PR TITLE
CEDS-1371: retrieving from new cache for items pages

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**

### CODE REVIEW

File: app/views/declaration/items_summary.scala.html [hunk 1]
```diff
@@ -56,38 +60,36 @@ <h1 id="title" style="margin:0">@messages("supplementary.itemsAdd.titleWithItems
...
-                            )
-                            </ol>
-                        </td>
-                    </tr>
-                }
+         @items.map { ei =>
+                <tr>
+                    <td scope="row">@ei.sequenceId</td>
+                    <td scope="row">@ei.procedureCodes.map(codes => codes.procedureCode.getOrElse("")) </td>
+                    <td scope="row">@ei.itemType.map(it => it.combinedNomenclatureCode) </td>
```
Comment (ed0906) [inline app/views/declaration/items_summary.scala.html]: `_.combinedNomenclatureCode`

File: app/views/declaration/items_summary.scala.html [hunk 2]
```diff
@@ -56,38 +60,36 @@ <h1 id="title" style="margin:0">@messages("supplementary.itemsAdd.titleWithItems
...
-                                <li>{a.quantity.getOrElse("")}</li>
-                            )
-                            </ol>
-                        </td>
-                    </tr>
-                }
+         @items.map { ei =>
+                <tr>
+                    <td scope="row">@ei.sequenceId</td>
+                    <td scope="row">@ei.procedureCodes.map(codes => codes.procedureCode.getOrElse("")) </td>
```
Comment (ed0906) [inline app/views/declaration/items_summary.scala.html]: `_.proceedureCode`

File: build.sbt [hunk 3]
```diff
@@ -58,18 +54,19 @@ lazy val microservice = Project(appName, file("."))
...
-    ,".*(BuildInfo|Routes|Options|TestingUtilitiesController).*"
+    "<empty>",
+    "Reverse.*",
+    "metrics\\..*",
+    "features\\..*",
+    "test\\..*",
+    ".*(BuildInfo|Routes|Options|TestingUtilitiesController).*"
   ).mkString(";"),
-  coverageMinimum := 81,
+  coverageMinimum := 87,
```
Comment (ed0906) [inline build.sbt]: :+1

File: app/controllers/declaration/ProcedureCodesPageController.scala [hunk 4]
```diff
@@ -53,16 +53,27 @@ class ProcedureCodesPageController @Inject()(
...
   def displayPage(itemId: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    customsCacheService.fetchAndGetEntry[ProcedureCodesData](goodsItemCacheId, formId).map {
-      case Some(data) =>
-        Ok(procedureCodesPage(appConfig, itemId, form.fill(data.toProcedureCode()), data.additionalProcedureCodes))
-      case _ => Ok(procedureCodesPage(appConfig, itemId, form, Seq()))
+    exportsCacheService.getItemByIdAndSession(itemId, journeySessionId).map {
+      case Some(exportItem) =>
+        exportItem.procedureCodes.fold({ Ok(procedureCodesPage(appConfig, itemId, form(), Seq())) }) {
+          procedureCodesData =>
+            Ok(
```
[the remark this answers is not in the document]
Comment (mattclark-zerogravit) [reply app/controllers/declaration/ProcedureCodesPageController.scala]: 👍

File: app/views/declaration/items_summary.scala.html [hunk 5]
```diff
@@ -56,38 +60,36 @@ <h1 id="title" style="margin:0">@messages("supplementary.itemsAdd.titleWithItems
...
+            @messages("site.add.item")
+        } else {
+            @messages("site.add.anotherItem")
         }
+        </a><p></p>
 
-
-        @if(items.nonEmpty) {
-            <a href="@WarehouseIdentificationController.displayForm().url" class="button" id="next">@messages("site.save_and_continue")</a>
+        @if(items.size > 0) {
```
[the remark this answers is not in the document]
Comment (mattclark-zerogravit) [reply app/views/declaration/items_summary.scala.html]: this crept back in probably during rebase... will fix in next PR. thank for pointing out

**Pull request merged**